test/byContract/IexecOrderManagement/IexecOrderManagement.ts

```diff
@@ -0,0 +1,393 @@
+            await iexecWrapper.signOrderOperation(orderOperation, datasetProvider);
+
+            await expect(iexecPoco.manageDatasetOrder(orderOperation))
+                .to.emit(iexecPoco, 'SignedDatasetOrder')
+                .withArgs(datasetOrderHash);
+            expect(await iexecPoco.viewPresigned(datasetOrderHash)).equal(datasetProvider.address);
+            expect(await iexecPoco.verifyPresignature(datasetProvider.address, datasetOrderHash)).is
+                .true;
+            expect(
+                await iexecPoco.verifyPresignatureOrSignature(
+                    datasetProvider.address,
+                    datasetOrderHash,
+                    NULL.SIGNATURE,
+                ),
+            ).is.true;
+        });
+        it('Should manage workerpool order when presign operation signed by workerpool provider', async () => {
+            const orderOperation = createOrderOperation(workerpoolOrder, OrderOperationEnum.SIGN);
+            await iexecWrapper.signOrderOperation(orderOperation, scheduler);
+
+            await expect(iexecPoco.manageWorkerpoolOrder(orderOperation))
+                .to.emit(iexecPoco, 'SignedWorkerpoolOrder')
+                .withArgs(workerpoolOrderHash);
+            expect(await iexecPoco.viewPresigned(workerpoolOrderHash)).equal(scheduler.address);
+            expect(await iexecPoco.verifyPresignature(scheduler.address, workerpoolOrderHash)).is
+                .true;
+            expect(
+                await iexecPoco.verifyPresignatureOrSignature(
+                    scheduler.address,
+                    workerpoolOrderHash,
+                    NULL.SIGNATURE,
+                ),
+            ).is.true;
+        });
+        it('Should manage request order when presign operation signed by requester', async () => {
+            const orderOperation = createOrderOperation(requestOrder, OrderOperationEnum.SIGN);
+            await iexecWrapper.signOrderOperation(orderOperation, requester);
+
+            await expect(iexecPoco.manageRequestOrder(orderOperation))
+                .to.emit(iexecPoco, 'SignedRequestOrder')
+                .withArgs(requestOrderHash);
+            expect(await iexecPoco.viewPresigned(requestOrderHash)).equal(requester.address);
+            expect(await iexecPoco.verifyPresignature(requester.address, requestOrderHash)).is.true;
+            expect(
+                await iexecPoco.verifyPresignatureOrSignature(
+                    requester.address,
+                    requestOrderHash,
+                    NULL.SIGNATURE,
+                ),
+            ).is.true;
+        });
+        it('Should match orders when presign operations signed by owners', async () => {
+            await depositInIexecAccounts();
+            await expect(
+                iexecPoco.matchOrders(appOrder, datasetOrder, workerpoolOrder, requestOrder),
+            ).to.emit(iexecPoco, 'OrdersMatched');
```

Comment:
Why do we want to test matchOrders in this file ?

Reply:
I did not include it in the first run but then when double checked I noticed:
https://github.com/iExecBlockchainComputing/PoCo/blob/904c4abd942fc2cad6fd9868aabc37db377a3c10/test/byContract/IexecOrderManagement/sign.js#L527
I prefer to include it in the review (for completeness, historical reasons, ..) then we can keep it or remove it if it is a total non-sense for you/us ( :ear::grey_question: )

Reply:
This comment https://github.com/iExecBlockchainComputing/PoCo/pull/101#discussion_r1706532906 also covers your question.

Reply:
It's feels more like IT than UT but why not, I would let @zguesmi give us his point of view

Reply:
I agree it's more of an IT to me.

Reply:
Yes, see https://github.com/iExecBlockchainComputing/PoCo/pull/101/commits/81f079d12c06abfafca123c0923ad96fda573e3d :)